src/EditorFeatures/CSharpTest/ReplaceDefaultLiteral/ReplaceDefaultLiteralTests.cs

```diff
@@ -409,78 +337,55 @@ void M()
         }
 
         [Theory]
-        [InlineData("int", "0")]
-        [InlineData("uint", "0U")]
-        [InlineData("byte", "0")]
-        [InlineData("sbyte", "0")]
-        [InlineData("short", "0")]
-        [InlineData("ushort", "0")]
-        [InlineData("long", "0L")]
-        [InlineData("ulong", "0UL")]
-        [InlineData("float", "0F")]
-        [InlineData("double", "0D")]
-        [InlineData("decimal", "0M")]
-        [InlineData("char", "'\\0'")]
-        [InlineData("string", "null")]
-        [InlineData("object", "null")]
-        public async Task TestCSharp7_1_InIsPattern_BuiltInType(string type, string expectedLiteral)
-        {
-            await TestWithLanguageVersionsAsync(
+        [InlineData("int")]
+        [InlineData("uint")]
+        [InlineData("byte")]
+        [InlineData("sbyte")]
+        [InlineData("short")]
+        [InlineData("ushort")]
+        [InlineData("long")]
+        [InlineData("ulong")]
+        [InlineData("float")]
+        [InlineData("double")]
+        [InlineData("decimal")]
+        [InlineData("char")]
+        [InlineData("string")]
+        [InlineData("object")]
+        public async Task TestCSharp7_1_InIsPattern_BuiltInType(string type)
+        {
+            await TestMissingWithLanguageVersionsAsync(
```

Comment:
Are we not able to do this fix any more because we don't have a target type for the 'default' expression? It feels like we should be able to fall back to the type of the LHS of the is-expression to synthesize a default value literal.

Would it be reasonable to instead disable the affected tests and add a follow-up issue?

Reply:
The current fixer focuses on making the type explicit on a valid `default`, but this scenario is invalid `default`. We could improve the error recovery in IDE logic to handle, but that doesn't seem a critical scenario. I'll let IDE folks advise.

---
In reply to: [314128753](https://github.com/dotnet/roslyn/pull/37596#discussion_r314128753) [](ancestors = 314128753)